src/App.tsx

```diff
@@ -1,39 +1,66 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
 
+import { Clock } from './components/Clock';
+
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
+
+export class App extends Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  // timerId = 0;
+
+  componentDidMount() {
+    // this.timerId = window.setInterval(() => {
+    window.setInterval(() => {
```

Comment:
```suggestion
    setInterval(() => {
```
You can omit using the global object.